src/knn.py

```diff
@@ -0,0 +1,56 @@
+"""Module for k nearest neighbor."""
+import numpy as np
+
+
+# K NEAREST NEIGHBOR (KNN)
+#
+# CodeFellows 401d5
+# Submission Date:
+#
+# Authors:  Colin Lamont <https://github.com/chamberi>
+#           Ben Shields <https://github.com/iamrobinhood12345>
+#
+# Methods:
+#
+#        predict(): predicts using the neares
+#        _get_winner(): finds the k closest classes, and returns the closest one to the predict function.
+
+
+class KNN(object):
+    """K Nearest Neighbor Classifier Object."""
+
+    def __init__(self, data, k=5):
+        """Initialize the KNN object."""
+        self.data = data
+        if k > len(data) or type(k) is not int or k <= 0:
+            raise ValueError("The number of neighbors to predict against must be an integer greater than 0 and less than size of data.")
+        self.k = 5
+
+    def predict(self, evals, k=None):
+        """Predict function that compares distances between eval data and training data."""
+        if not k:
+            k = self.k
+        win = []
+        for eval_item in evals:        # get each set of eval
+            distance_list = []
+            for data_idx in range(len(self.data)):
+                squares_sum = 0.0
+                for item_idx in range(len(eval_item)):
+                    squares_sum += (eval_item[item_idx] - self.data[data_idx][item_idx]) ** 2
```

Comment:
You're missing out on the power of numpy or pandas to perform math operations across an entire data set. The distance calculation could be one-lined.